stdlib/public/stubs/Stubs.cpp

```diff
@@ -128,6 +128,26 @@ static locale_t getCLocale() {
 }
 #endif
 
+#ifdef __ANDROID__
+#define uselocale(a) (nullptr)
+static double swift_strtod_l(const char* a, char** b, locale_t c) {
+	return strtod(a, b);
+}
+static float swift_strtof_l(const char* a, char** b, locale_t c) {
+	return strtof(a, b);
```

Comment:
@gribozavr On api 21 (Android 5.0) uselocale is present, so the real uselocale can be used. However, strtod_l and strtof_l are not present on Android even in the latest version (6.0), and strtold_l is also missing in 5.0 (the version we're targetting). So I don't think locale support is possible at this time.